overlord/hookstate/context_test.go

```diff
@@ -49,11 +53,40 @@ func (s *contextSuite) SetUpTest(c *C) {
 	var err error
 	s.context, err = NewContext(s.task, s.task.State(), s.setup, nil, "")
 	c.Check(err, IsNil)
+
+	s.componentTask = s.state.NewTask("test-component-task", "my test component task")
+	s.componentSetup = &HookSetup{
+		Snap:              "test-snap",
+		Revision:          snap.R(1),
+		Hook:              "test-hook",
+		Component:         "test-component",
+		ComponentRevision: snap.R(1),
```

Comment:
probably a good idea to use a different value here, otherwise we want catch errors were the fields get confused with each other